tutorials/arvo/ames.md

```diff
@@ -112,40 +115,299 @@ ducts as unique keys.  Stop tracking a peer by sending Ames a
+`~worwel-sipnum`. First, it [`+cue`s](@/docs/reference/library/2p.md#cue)
+(deserializes) the payload and changes the `origin` to the IP and port of
+`~bacbel-tagfeb`. Then it `+jam`s `[origin content]` to form a new payload, and
+`+mug`s that payload to get a new checksum. It replaces the old payload and
+checksum with the new, and then sends the packet along to `~worwel-sipnum`.
+
+Once `~worwel-sipnum` processes the packet, it will know the IP and port of
+`~bacbel-tagfeb` since `~zod` included it when it forwarded the packet and so
+will send an ack packet directly to `~bacbel-tagfeb`. Communication between the
+two ships will now be direct until one of them changes their IP address, port, or networking keys.
+
+
+## The Serf and the King
+
+Urbit's functionality is split between the two binaries `urbit-worker` (sometimes
+called the Serf) and `urbit-king` (sometimes called the King). This division of
+labor is currently not well-documented outside of the [Vere documents](@/docs/tutorials/vere/_index.md), but we summarize it here.
+
+In short, the Serf is the Nock runtime and so keeps track of the current state
+of Arvo as a Nock noun and updates the state by `%poke`ing it with nouns, and
+then informs the King of the new state. The King manages snapshots of the Arvo
+state and handles I/O with Unix, among other things. The Serf only ever talks to
+the King, while the King talks with both the Serf and Unix.
+
+### Ames
+
+The King has several submodules, one of them being an Ames I/O submodule. This
+submodule is responsible for wrapping outgoing Ames packets as UDP packets, and
+looking at incoming UDP packets handed to it by Unix to ensure that they are
+valid Ames packets. It also maintains an incoming and outgoing packet queue.
+
+This division is summarized in the following diagram, describing how
+`~bacbel-tagfeb` requests a subscription to the `recipes` notebook of
+`~worwel-sipnum` in the Publish app.
+
+<div style="text-align:center">
+<img src="https://media.urbit.org/docs/arvo/packet.png">
+</div>
+
+Ames, as a part of Arvo, handles `+jam`ming, packetizing, encryption, and
+forming Ames packets. Once it is ready to send an Ames packet, it `%give`s to
+Unix a `%send` `gift` containing that packet. This will be a Nock noun
+containing the `@tas` `%send` as well as the serialized packet.
+
+"Unix", in this case, is actually the King. The King receives the `%send`
+instruction, wraps the packet contained within as a UDP packet, and queues the
+UDP packet contained. Once it is time to send it (which will be determined by
+congestion control), it will go ahead and forward the packet to Unix's
+networking interface, at which point it is no longer controlled by any part of Urbit.
+
+Now the receiving King is handed a UDP packet by Unix. The King removes the UDP
+wrapped and checks to see that it has received a valid Ames packet. If so, it
+`+cue`s the payload and wraps it as a `%hear` `note` and passes it to the Serf.
```

Comment:
The daemon doesn't `+cue` the payload; it jams the lane on which it heard the packet, but other than that, it delivers the packet to the Ames vane as an atom by coping the bytes it heard on its UDP port.